java/src/org/openqa/selenium/support/decorators/WebDriverDecorator.java

```diff
@@ -336,9 +346,22 @@ protected final <Z> Z createProxy(final Decorated<Z> decorated, Class<Z> clazz)
       .getLoaded()
       .asSubclass(clazz);
 
+    if (clazz.isInterface() || this.targetWebDriverConstructorArgs.length == 0) {
+      try {
+        return proxy.newInstance();
+      } catch (ReflectiveOperationException e) {
+        throw new IllegalStateException("Unable to create new proxy", e);
+      }
+    }
+
     try {
-      return proxy.newInstance();
-    } catch (ReflectiveOperationException e) {
+      Constructor<?> dstConstructor = proxy
+        .getConstructor(Arrays.stream(targetWebDriverConstructorArgs)
+          .map(Object::getClass)
+          .toArray(Class<?>[]::new)
+        );
```

Comment:
That might not work in 100% cases, for example: constructor arg type is `List`, but argument can be `ArrayList`. And for such cases fully automated search for applicable constructor is not a trivial task. So I would recommend to overload `WebDriverDecorator` constructor and add one more field:

```java
  private final Class<T> targetWebDriverClass;
  private final Class<?>[] targetWebDriverConstructorArgsTypes;
  private final Object[] targetWebDriverConstructorArgs;

  private Decorated<T> decorated;

...

  public WebDriverDecorator(Class<T> targetClass) {
    this(targetClass, new Object[0]);
  }

  public WebDriverDecorator(Class<T> targetClass, Object[] targetConstructorArgs) {
    this.targetWebDriverClass = targetClass;
    this.targetWebDriverConstructorArgs = targetConstructorArgs;
    this.targetWebDriverConstructorArgsTypes = Arrays.stream(targetWebDriverConstructorArgs)
          .map(Object::getClass)
          .toArray(Class<?>[]::new);
  }

  public WebDriverDecorator(Class<T> targetClass, Class<?>[] targetWebDriverConstructorArgsTypes, Object[] targetConstructorArgs) {
    this.targetWebDriverClass = targetClass;
    this.targetWebDriverConstructorArgsTypes = targetWebDriverConstructorArgsTypes;
    this.targetWebDriverConstructorArgs = targetConstructorArgs;
  ...
```